src/core/ext/transport/chttp2/transport/flow_control.cc

```diff
@@ -351,6 +353,33 @@ FlowControlAction::Urgency TransportFlowControl::DeltaUrgency(
   }
 }
 
+namespace {
+
+struct WindowDelta {
+  int64_t min = 0;
+  int64_t max = 0;
+};
+
+void StreamLocalWindowDeltaCallback(void* user_data, uint32_t /*key*/,
+                                    void* stream) {
+  WindowDelta* delta = static_cast<WindowDelta*>(user_data);
+  grpc_chttp2_stream* s = static_cast<grpc_chttp2_stream*>(stream);
+  delta->min = GPR_MIN(delta->min, s->flow_control->local_window_delta());
+  delta->max = GPR_MAX(delta->max, s->flow_control->local_window_delta());
+}
+
+// Loops over all streams in the transport and returns WindowDelta where max has
+// the highest window delta from all the streams and min has the lowest window
+// delta. Min has a ceiling of 0, while max has a floor of 0.
+WindowDelta ComputeLocalWindowDeltaBoundaries(grpc_chttp2_transport* t) {
+  WindowDelta delta;
+  grpc_chttp2_stream_map_for_each(&t->stream_map,
```

Comment:
does this change the update from O(1) to O(N)?

Reply:
Yes it does unfortunately. I wonder if there is a way to do this without adding an O(N) loop

Reply:
I wonder too... as it is this feels a little risky in terms of CPU usage.